src/schema/sources.ts

```diff
@@ -347,56 +375,131 @@ const sourceToGQL = (source: Source): GQLSource => ({
 });
 
 export enum SourcePermissions {
-  View,
-  Post,
-  Leave,
-  Delete,
-  Edit,
+  View = 'view',
+  Post = 'post',
+  PostDelete = 'post_delete',
+  RemoveMember = 'remove_member',
+  Leave = 'leave',
+  Delete = 'delete',
+  Edit = 'edit',
+}
+
+const canRemoveMemberRoles = [
+  SourceMemberRoles.Owner,
+  SourceMemberRoles.Moderator,
+];
+
+const canPostDeleteRoles = [
+  SourceMemberRoles.Owner,
+  SourceMemberRoles.Moderator,
+];
```

Comment:
Really agree with that. Let me change the one for members so it will be easier to manage if there will be numerous actions under the same context.

Wdyt of below examples:
- member remove
- post delete

So when we can add directly it will be:
- member add

Reply:
That makes sense, let's do that 🙏

Reply:
Instead of having multiple arrays per action, I suggest having an array of actions per role.
Otherwise there will be more work to add new action or role.

Reply:
That would mean we will no longer use the switch case since we will explicitly define their access. Let me update the current implementation quickly.